File: examples/complex_ts_project_mode/generated_project/src/rules/discounts.ts
import { toMoney } from "../pricing/rounding.js";

export function loyaltyDiscount(subtotal: number, isLoyalCustomer: boolean): number {
  if (!isLoyalCustomer) return 0;
  return toMoney(Math.min(subtotal * 0.05, 25));
}

export function bulkUnitsDiscount(totalUnits: number): number {
  if (totalUnits >= 20) return 10;
  if (totalUnits >= 10) return 4;
  return 0;
}

File: examples/complex_project_mode/generated_project/app/rules/discounts.py
from __future__ import annotations


from .taxing import to_money


def loyalty_discount(subtotal: float, *, is_loyal_customer: bool) -> float:
    if not is_loyal_customer:
        return 0.0
    return to_money(min(subtotal * 0.05, 25.0))


def bulk_units_discount(total_units: int) -> float:
    if total_units >= 20:
        return 10.0
    if total_units >= 10:
        return 4.0
    return 0.0
